frontend/request.go

```diff
@@ -137,6 +138,10 @@ func marshalDockerfile(ctx context.Context, dt []byte, opts ...llb.ConstraintsOp
 
 func MaybeSign(ctx context.Context, client gwclient.Client, st llb.State, spec *dalec.Spec, targetKey string) (llb.State, error) {
 	if signingDisabled(client) {
+		err := Warn(ctx, client, st, "Signing disabled by build-arg "+keySkipSigningArg)
+		if err != nil {
+			bklog.G(ctx).WithError(err).Error("Error writing warning log")
```

Comment:
We could be a bit more specific about being an error creating the warning about the signing being disabled. Might help in the future if we add more warning?

Reply:
I'm not sure I 100% follow here, but I've updated this to move all the error handling in the `Warn` function and also made it more robust to such errors.